src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2709,6 +2716,17 @@ FixedwingPositionControl::tecs_update_pitch_throttle(const float control_interva
 		     hgt_rate_sp);
 
 	tecs_status_publish(alt_sp, airspeed_sp, airspeed_rate_estimate, throttle_trim_adjusted);
+
+	// Estimate if we are in level flight.
+	// Tecs must be running, the height rate setpoint is not set or set to 0 and we are close to the target altitude and target altitude is not moving
+	if (_tecs_is_running && (!PX4_ISFINITE(hgt_rate_sp) || fabsf(hgt_rate_sp) < 0.1f)
+	    && fabsf(_current_altitude - alt_sp) < _param_nav_fw_alt_rad.get()
+	    && fabsf((alt_sp - _last_tecs_alt_sp) / control_interval) < 0.1f) {
```

Comment:
If we decide to keep these artificial thresholds: please make them constants.

Reply:
I will include that. Do you have any better value for the threshholds? I basically want to say that the feedforward height rate and the height setpoint changes should be around zero, hence i just chose 0.1. Open for more suggestions. I don't think this requires a parameter so i will make a constant out of it.